session.go

```diff
@@ -333,12 +334,12 @@ runLoop:
 		s.handshakeChan <- handshakeEvent{err: closeErr.err}
 	}
 	s.handleCloseError(closeErr)
-	close(s.runClosed)
+	s.ctxCancel()
```

Comment:
Well, it's literally one line above the `return`, but I'm fine with this if it makes you feel safer.